File: modules/master-data/Product/Entities/Product.cs
using System.ComponentModel.DataAnnotations;

namespace ERP.Modules.MasterData.Product.Entities;

public class Product
{
    public Guid Id { get; set; } = Guid.NewGuid();

    [Required]
    [MaxLength(50)]
    public string Code { get; set; } = string.Empty;

    [Required]
    [MaxLength(200)]
    public string Name { get; set; } = string.Empty;

    [MaxLength(1000)]
    public string? Description { get; set; }

    [MaxLength(100)]
    public string? Category { get; set; }

    [MaxLength(50)]
    public string? Brand { get; set; }

    [MaxLength(50)]
    public string? Unit { get; set; }

    public decimal Price { get; set; }

    public decimal CostPrice { get; set; }

    public int StockQuantity { get; set; }

    public int MinimumStock { get; set; }

    public bool IsActive { get; set; } = true;

    [MaxLength(500)]
    public string? ImageUrl { get; set; }

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public DateTime? UpdatedAt { get; set; }

    [MaxLength(100)]
    public string CreatedBy { get; set; } = string.Empty;

    [MaxLength(100)]
    public string? UpdatedBy { get; set; }
}

File: ui/react-app/src/types/product.ts
export interface Product {
  id: string;
  code: string;
  name: string;
  description?: string;
  category?: string;
  brand?: string;
  unit?: string;
  price: number;
  costPrice: number;
  stockQuantity: number;
  minimumStock: number;
  isActive: boolean;
  imageUrl?: string;
  createdAt: string;
  updatedAt?: string;
  createdBy: string;
  updatedBy?: string;
}

export interface CreateProductRequest {
  code: string;
  name: string;
  description?: string;
  category?: string;
  brand?: string;
  unit?: string;
  price: number;
  costPrice: number;
  stockQuantity: number;
  minimumStock: number;
  isActive: boolean;
  imageUrl?: string;
  createdBy: string;
}

export interface UpdateProductRequest {
  name: string;
  description?: string;
  category?: string;
  brand?: string;
  unit?: string;
  price: number;
  costPrice: number;
  stockQuantity: number;
  minimumStock: number;
  isActive: boolean;
  imageUrl?: string;
  updatedBy: string;
}